benchmark/single-source/StringReplaceSubrange.swift

```diff
@@ -0,0 +1,116 @@
+//===--- StringReplaceSubrange.swift -------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+let tags: [BenchmarkCategory] = [.validation, .api, .String]
+
+public let StringReplaceSubrange = [
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.SmallLiteral.String",
+    runFunction: { replaceSubrange($0, "coffee", with: "t") },
+    tags: tags
```

Comment:
Would you mind explaining what is the difference between the small literal string vs the large managed string? Is this something related to [this small string optimization](https://gist.github.com/milseman/bb39ef7f170641ae52c13600a512782f#small-string-optimization)? If the string fits 15 ASCII characters length, it won't be allocated in the heap memory?

Reply:
Correct. See: [`_SmallString`](https://github.com/apple/swift/blob/master/stdlib/public/core/SmallString.swift) and  [`_StringGuts`](https://github.com/apple/swift/blob/master/stdlib/public/core/StringGuts.swift) for implementation details if you're interested.

Reply:
Thanks for the link! I will take a look at them 🔍

Reply:
Small form accommodates 15 UTF-8 code units in length (not just ASCII)

Reply:
Thank you for clarifying the length of the small string, Michael 😄